aten/src/ATen/TensorImpl.h

```diff
@@ -17,11 +17,35 @@ struct Storage;
 struct Tensor;
 } // namespace at
 
+namespace at {
+struct TensorImplOptions {
+  bool has_storage_;
+  bool has_strides_;
+  // Whether this type of tensor supports changing dimensionality via maybe_zero_dim
+  bool support_resize_by_maybe_zero_dim_;
```

Comment:
honestly I don't think it's worth having this be separate; no one will know what to set it to in any case.  We only call this from generated code into TH, which has to be a dense tensor.  Just check has_storage_ && has_strides_ and put an explanation in the comments.